lib/devicetrust/native/device_linux.go

```diff
@@ -51,34 +51,64 @@ var linuxDevice = &tpmDevice{
 }
 
 func enrollDeviceInit() (*devicepb.EnrollDeviceInit, error) {
-	return linuxDevice.enrollDeviceInit()
+	init, err := linuxDevice.enrollDeviceInit()
+	return init, rewriteTPMPermissionError(err)
 }
 
 func signChallenge(chal []byte) (sig []byte, err error) {
-	return linuxDevice.signChallenge(chal)
+	return nil, errors.New("signChallenge not implemented for TPM devices")
 }
 
 func getDeviceCredential() (*devicepb.DeviceCredential, error) {
-	return linuxDevice.getDeviceCredential()
+	cred, err := linuxDevice.getDeviceCredential()
+	return cred, rewriteTPMPermissionError(err)
 }
 
 func solveTPMEnrollChallenge(
 	chal *devicepb.TPMEnrollChallenge,
 	debug bool,
 ) (*devicepb.TPMEnrollChallengeResponse, error) {
+	// No need to call rewriteTPMPermissionError here, enrollDeviceInit must pass
+	// first.
 	return linuxDevice.solveTPMEnrollChallenge(chal, debug)
 }
 
 func solveTPMAuthnDeviceChallenge(
 	chal *devicepb.TPMAuthenticateDeviceChallenge,
 ) (*devicepb.TPMAuthenticateDeviceChallengeResponse, error) {
-	return linuxDevice.solveTPMAuthnDeviceChallenge(chal)
+	resp, err := linuxDevice.solveTPMAuthnDeviceChallenge(chal)
+	return resp, rewriteTPMPermissionError(err)
 }
 
 func handleTPMActivateCredential(encryptedCredential, encryptedCredentialSecret string) error {
 	return errors.New("elevated credential activation not implemented for linux")
 }
 
+func rewriteTPMPermissionError(err error) error {
+	// We are looking for an error that looks roughly like this:
+	//
+	// 	err = &fs.PathError{
+	// 		Path: "/dev/tpmrm0",
+	// 		Err: fs.ErrPermission,
+	// 	}
+	if !errors.Is(err, fs.ErrPermission) {
+		return err
+	}
+
+	pathErr := &fs.PathError{}
+	if !errors.As(err, &pathErr) || pathErr.Path != "/dev/tpmrm0" {
+		return err
+	}
+	log.
+		WithError(err).
+		Debug("TPM: Replacing TPM permission error with a more friendly one")
+
+	return errors.New("" +
+		"Failed to open the TPM device. " +
+		"Consider assigning the user to the `tss` group or creating equivalent udev rules. " +
+		"See https://goteleport.com/docs/access-controls/device-trust/device-management/#troubleshooting.")
```

Comment:
Note: I'll update this section with instructions shortly.

Reply:
Is the `tss` group a standard thing? Should we check the owner of the device to make sure?

Reply:
As far as I can tell, yes, it exists in my Ubuntu install by default. I believe it gets created along with TPM tools: https://github.com/tpm2-software/tpm2-tss/tree/master#overview.

Not sure what you meant by checking the device owner?

Reply:
A better link about tss: https://github.com/tpm2-software/tpm2-tss/blob/master/dist/tpm-udev.rules#L4.